gapic/src/main/com/google/gapid/views/TextureList.java

```diff
@@ -0,0 +1,551 @@
+        createTableColumn(viewer, "Format", Data::getFormat,
+            Comparator.comparing(Data::getFormat)));
+  }
+
+  @Override
+  public Control getControl() {
+    return this;
+  }
+
+  @Override
+  public void reinitialize() {
+    onCaptureLoadingStart(false);
+    updateTextures(true);
+  }
+
+  @Override
+  public void onCaptureLoadingStart(boolean maintainState) {
+    clear();
+  }
+
+  @Override
+  public void onCaptureLoaded(Loadable.Message error) {
+    if (error != null) {
+      clear();
+    }
+  }
+
+  @Override
+  public void onResourcesLoaded() {
+    updateTextures(true);
+  }
+
+  @Override
+  public void onCommandsSelected(CommandIndex path) {
+    updateTextures(false);
+  }
+
+  @Override
+  public void onTextureFollowed(Service.Resource resource) {
+    TableItem[] items = textureTable.getTable().getItems();
+    for (int i = 0; i < items.length; i++) {
+      Data d = (Data)(items[i].getData());
+      if (d.info.getID().equals(resource.getID())) {
+        textureTable.getTable().setSelection(items[i]);
+        break;
+      }
+    }
+  }
+
+  private void updateTextures(boolean resourcesChanged) {
+    if (!models.resources.isLoaded()) {
+      // TODO: show capture loading message
+    } else if (models.commands.getSelectedCommands() == null) {
+      // TODO: show Messages.SELECT_COMMAND
+      clear();
+    } else {
+      // Memorize selection index before disposing image resource.
+      // When comparator is reset, the table is refreshed, and early image disposal will introduce null error.
+      ViewerComparator comparator = textureTable.getComparator();
+      textureTable.setComparator(null);
+      int selection = textureTable.getTable().getSelectionIndex();
+
+      imageProvider.reset();
+
+      Widgets.Refresher refresher = withAsyncRefresh(textureTable);
+      List<Data> textures = Lists.newArrayList();
+      models.resources.getResources(TextureResource).stream()
+          .map(r -> new Data(r.resource, r.deleted))
+          .forEach(data -> {
+            textures.add(data);
+            data.load(models.resources, textureTable.getTable(), refresher);
+          });
+
+      textureTable.setInput(textures);
+      packColumns(textureTable.getTable());
+
+      if (!resourcesChanged && selection >= 0 && selection < textures.size()) {
+        textureTable.getTable().select(selection);
+      }
+      textureTable.setComparator(comparator);
+      updateSelection();
+    }
+  }
+
+  private void clear() {
+    textureTable.setInput(Collections.emptyList());
+    textureTable.getTable().requestLayout();
+    imageProvider.reset();
+  }
+
+  private void updateSelection() {
+    int selection = textureTable.getTable().getSelectionIndex();
+    if (selection >= 0) {
+      Data data = (Data)textureTable.getElementAt(selection);
+      Path.ResourceData path = models.resources.getResourcePath(data.info);
+      models.follower.onFollow(Path.Any.newBuilder().setResourceData(path).build());
```

Comment:
I wonder if it makes more sense to manage this via the resource model - i.e. have something like `Resources.selectTexture`, etc. but maybe for now this is ok.

Reply:
I agree, implemented as discussed.